### PR TITLE
Database: clarify list of  SQL `db.system`s

### PR DESCRIPTION
**Pull request by lmolkova**
## Changes

Following up on the WG discussion inspired by  https://github.com/open-telemetry/semantic-conventions/issues/965

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

File: docs/database/sql.md [hunk 1]
```diff
@@ -6,9 +6,34 @@ linkTitle: SQL
...
 **Status**: [Experimental][DocumentStatus]
 
-The SQL databases Semantic Conventions extend and override the [Database Semantic Conventions](database-spans.md)
-that describe common database operations attributes in addition to the Semantic Conventions
-described on this page.
+The SQL databases Semantic Conventions describes how common [Database Semantic Conventions](database-spans.md) apply to SQL databases.
+
+The following database systems (defined in the [`db.system`](./database-spans.md#notes-and-well-known-identifiers-for-dbsystem) set) are known to use SQL as their primary query language:
+
+- `cockroachdb`
```
Comment (lmolkova) [inline docs/database/sql.md:13]: @jsuereth here's an example (tagging SQL databases) where enum tags would be great

File: docs/database/sql.md [hunk 2]
```diff
@@ -6,9 +6,34 @@ linkTitle: SQL
...
+- `oracle`
+- `pervasive`
+- `postgresql`
+- `sqlite`
+- `trino`
+- `other_sql`
+
+Many other database systems provide SQL support and can be accessed via generic database driver APIs such as JDBC or ODBC.
+When system-specific conventions are defined for the database, and it's accessed using SQL, the instrumentation SHOULD
+follow database-specific conventions and MAY also adhere to SQL semantic conventions.
```
Comment (trask) [inline docs/database/sql.md:36]: this seems a little confusing, e.g. for `mssql` we are copying down all of the applicable sql semantic conventions to the system-specific convention

and for some other system, say cosmosdb, if it's accessed using jdbc, I think I would expect it to follow SQL semantic conventions (and not necessarily cosmosdb conventions)

Comment (lmolkova) [reply docs/database/sql.md:36]: > mssql we are copying down all of the applicable sql semantic conventions to the system-specific convention

let me think how to make it more clear.

> and for some other system, say cosmosdb, if it's accessed using jdbc, I think I would expect it to follow SQL semantic conventions (and not necessarily cosmosdb conventions)

It's interesting.

E.g. with Cosmos users are still using CosmosDB SDK to query with SQL [e.g. this](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/quickstart-java?pivots=devcontainer-codespace#query-items).

[Mongodb-jdbc driver](https://mvnrepository.com/artifact/org.mongodb/mongodb-jdbc/2.1.2) or [cassandra unofficial  jdbc driver](https://github.com/zhicwu/cassandra-jdbc-driver/tree/master) work on top of corresponding Mongo and Cassandra client libs.  If the underlying libraries are instrumented, the driver does not need to be, i.e. SQL conventions should not be expected.

If I understand correctly, your concern is for a different case when the driver is instrumented.

Do you think the following would work for both of our cases:

> Specific database client libraries SHOULD follow corresponding semantic conventions (if defined) and MAY also follow SQL conventions when database is being accessed via SQL.

> Instrumentations applied to generic SQL abstractions such as JDBC have limited knowledge about underlying databases and SHOULD adhere to SQL semantic conventions.

File: .chloggen/1024.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+change_type: enhancement
+component: db
+note: List SQL database systems one the SQL semantic conventions.
```
Comment (joaopgrassi) [inline .chloggen/1024.yaml:3]:
```suggestion
note: List SQL database systems on the SQL semantic conventions.
```

File: docs/database/sql.md [hunk 4]
```diff
@@ -6,9 +6,33 @@ linkTitle: SQL
 
 **Status**: [Experimental][DocumentStatus]
 
-The SQL databases Semantic Conventions extend and override the [Database Semantic Conventions](database-spans.md)
-that describe common database operations attributes in addition to the Semantic Conventions
-described on this page.
+The SQL databases Semantic Conventions describes how common [Database Semantic Conventions](database-spans.md) apply to SQL databases.
+
+The following database systems (defined in the [`db.system`](./database-spans.md#notes-and-well-known-identifiers-for-dbsystem) set) are known to use SQL as their primary query language:
```
Comment (joaopgrassi) [inline docs/database/sql.md:11]: hum, could we maybe use `tags` to generate this list automatically?

File: docs/database/sql.md (hunk 2 above)
Comment (trask) [reply docs/database/sql.md]: > Instrumentations applied to specific database client libraries SHOULD follow corresponding database semantic conventions (if defined) and MAY also follow SQL conventions when database is being accessed via SQL.

can you give an example what this means in practice? e.g. how would a particular comsosdb span change if it was also following SQL conventions in [e.g. this case](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/quickstart-java?pivots=devcontainer-codespace#query-items).[e.g. this](https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/quickstart-java?pivots=devcontainer-codespace#query-items)

> Instrumentations applied to generic SQL abstractions such as JDBC have limited knowledge about underlying databases and SHOULD adhere to SQL semantic conventions.

I like this clarification

File: docs/database/sql.md (hunk 4 above)
Comment (lmolkova) [reply docs/database/sql.md:11]: not possible today :(

**Review by trask — APPROVED** [review]

File: docs/database/sql.md [hunk 5]
```diff
@@ -6,9 +6,33 @@ linkTitle: SQL
...
+- `maxdb`
+- `mssql`
+- `mssqlcompact`
+- `mysql`
+- `oracle`
+- `pervasive`
+- `postgresql`
+- `sqlite`
+- `trino`
+- `other_sql`
```
Comment (trask) [inline docs/database/sql.md:32]: when we stabilize, will want to split this into two sections one stable and one experimental

File: docs/database/sql.md [hunk 6]
```diff
@@ -6,9 +6,33 @@ linkTitle: SQL
...
+
+The following database systems (defined in the [`db.system`](./database-spans.md#notes-and-well-known-identifiers-for-dbsystem) set) are known to use SQL as their primary query language:
+
+- `cockroachdb`
+- `db2`
+- `derby`
+- `edb`
+- `ingres`
+- `interbase`
+- `firebird`
```
Comment (lmolkova) [inline docs/database/sql.md:19]: todo: fix the order

**Review by gregkalapos — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Pull request merged**